File: implementacion_java/src/utilidades/BorderRoundedStroke.java
package utilidades;

import java.awt.*;
import java.awt.geom.RoundRectangle2D;

public class BorderRoundedStroke extends BasicStroke {
    public BorderRoundedStroke(float width) {
        super(width);
    }

    @Override
    public Shape createStrokedShape(Shape shape) {
        Shape s = super.createStrokedShape(shape);
        return new RoundRectangle2D.Float(
                s.getBounds().x, s.getBounds().y,
                s.getBounds().width, s.getBounds().height, 10f, 10f
        );
    }
}

File: implementacion_kotlin/src/utilidades/BorderRoundedStroke.kt
package utilidades

import java.awt.BasicStroke
import java.awt.Shape
import java.awt.geom.RoundRectangle2D

class BorderRoundedStroke(width: Float) : BasicStroke(width) {
    override fun createStrokedShape(shape: Shape): Shape {
        val s = super.createStrokedShape(shape)
        return RoundRectangle2D.Float(
            s.bounds.x.toFloat(), s.bounds.y.toFloat(),
            s.bounds.width.toFloat(), s.bounds.height.toFloat(), 10f, 10f
        )
    }
}
